lib/board.rb

```diff
@@ -0,0 +1,86 @@
+
+
+class Board
+	attr_accessor :cells, :answer_token
+
+	def initialize
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+		@answer_token = []
+	end
+
+	def reset!
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+	end
+
+	def display
+		puts ""
+		puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+		puts "-----------"  
+		puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+		puts "-----------" 
+		puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+		puts ""
+	end
+
+	def which_cornors?
+		corners = ["1", "3", "7", "9"]
+		eligible_corners = corners.select do |corner|
+			!self.taken?(corner)
+		end
+		eligible_corners.sample
+	end
+
+	def available_spots
+		available = []
+		self.cells.each_with_index do |cell, index|
+			if cell == " "
+				available << (index + 1).to_s
+			end
+		end
+		available.sample
+	end
+
+	def block_positions
+		board = self.cells
+		@answer_token = []
+		answers = []
+		Game::WIN_COMBINATIONS.each do |c|
+			indexes = [c[0], c[1], c[2]]
+			my_combo = [board[c[0]], board[c[1]], board[c[2]]]
+			if  my_combo.sort === [" ", "X", "X"] || my_combo.sort === [" ", "O", "O"]
+				answer_token << my_combo.sort[2]
+				answers << indexes.select do |index|
+					board[index] == " "
+				end	
+			end
+		end
+		answers.flatten!
+		answers = answers.map { |a| a + 1}
+		answers
+	end
+
+	def position(position)
+		@cells[position.to_i - 1]
+	end
+
+	def full?
+		!self.cells.include?(" ")
```

Comment:
Consider using the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.